### ISSUE TITLE
LabelledStatement should allow only Statement or FunctionDeclaration, not ClassDeclaration or LexicalDeclaration

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#193]
legal:
```javascript
mylabel: var x;
otherlabel: function f() {}
```

illegal:
```javascript
mylabel: const x = 3;
otherlabel: let y;
thirdlabel: class C {}
```

quick-lint-js thinks all of the above examples are legal.

We already check for `const`, `let`, and `class` in bodies of `with` statements. We could probably reuse that code for labels.

**Comment by Guilherme-Vasconcelos** [comment]
May I get assigned to this issue? I'd like to give it a try

**Comment by strager** [comment]
@Guilherme-Vasconcelos Sure. Let me know if you need any suggestions.
